src/main/java/io/github/moulberry/notenoughupdates/overlays/EquipmentOverlay.java

```diff
@@ -378,6 +383,70 @@ public void renderPets(GuiInventory inventory, int mouseX, int mouseY, int width
 		}
 	}
 
+	private final Map<ItemStack, Integer> itemsToAdd = new HashMap<>();
+
+	@SubscribeEvent
+	public void onClickItem(PlayerInteractEvent event) {
+		if (event.action == PlayerInteractEvent.Action.LEFT_CLICK_BLOCK || Minecraft.getMinecraft().thePlayer.getHeldItem() == null) return;
+
+		ItemStack heldItem = Minecraft.getMinecraft().thePlayer.getHeldItem();
+		NBTTagList heldItemLore = heldItem.getTagCompound().getCompoundTag("display").getTagList("Lore", 8);
+
+		String itemType = Objects.requireNonNull(StringUtils.substringAfterLast(heldItemLore.get(heldItemLore.tagCount() - 1).toString(), " "), "null").replace("\"", "");
+		if (!Arrays.asList("NECKLACE", "CLOAK", "BELT", "GLOVES", "BRACELET").contains(itemType)) return;
+
+		NEUConfig.HiddenProfileSpecific profileSpecific = NotEnoughUpdates.INSTANCE.config.getProfileSpecific();
+		if (profileSpecific == null) return;
+
+		int slot;
+
+		switch (itemType) {
+			case "NECKLACE":
+				slot = 10;
+				break;
+			case "CLOAK":
+				slot = 19;
+				break;
+			case "BELT":
+				slot = 28;
+				break;
+			case "GLOVES": case "BRACELET":
+				slot = 37;
+				break;
+			default:
+				return;
+		}
+
+		ItemStack itemInSlot = NotEnoughUpdates.INSTANCE.manager.jsonToStack(profileSpecific.savedEquipment.get(slot).getAsJsonObject(), false);
+		if (itemInSlot != null && itemInSlot.getDisplayName().contains("Empty")) itemsToAdd.put(heldItem, slot);
+	}
+
+	@SubscribeEvent
+	public void onReceiveChatMessage(ClientChatReceivedEvent event) {
+		if (event.type == 2 || !event.message.getUnformattedText().startsWith("You equipped a ") || itemsToAdd.isEmpty()) return;
+
+		try {
+			for (ItemStack item : itemsToAdd.keySet()) {
+				if (event.message.getUnformattedText().contains(net.minecraft.util.StringUtils.stripControlCodes(item.getDisplayName()))) {
+
+					NEUConfig.HiddenProfileSpecific profileSpecific = NotEnoughUpdates.INSTANCE.config.getProfileSpecific();
+					if (profileSpecific == null) return;
+
+					JsonObject itemToSave = NotEnoughUpdates.INSTANCE.manager.getJsonForItem(item);
+					if (!itemToSave.has("internalname")) {
+						itemToSave.add("internalname", new JsonPrimitive("_"));
+					}
```

Comment:
Maybe copy that item stack enriching logic into a shared method so the other code path shares this logic.